jvector-base/src/main/java/io/github/jbellis/jvector/graph/disk/OnDiskGraphIndexWriter.java

```diff
@@ -155,6 +204,86 @@ public synchronized void write(Map<FeatureId, IntFunction<Feature.State>> featur
 
         writeHeader(view); // sets position to start writing features
 
+        // Write L0 records (either parallel or sequential)
+        if (useParallelWrites) {
+            writeL0RecordsParallel(featureStateSuppliers);
+        } else {
+            writeL0RecordsSequential(view, featureStateSuppliers);
+        }
+
+        // We will use the abstract method because no random access is needed
+        writeSparseLevels(view);
+
+        // We will use the abstract method because no random access is needed
+        writeSeparatedFeatures(featureStateSuppliers);
+
+        // Write the header again with updated offsets
```

Comment:
Did you mean to move this comment to line 225, and here say "write footer again"?